Improve pie chart error messages

costaparas opened the pull request:
## PR summary

Fix typo in error message, add more detail and make formatting consistent

## PR checklist
<!-- Please mark any checkboxes that do not apply to this PR as [N/A].-->

- [na] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [na] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [na] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

rcomer reviewed (approved):
Thanks @costaparas!  I agree this is better.

costaparas commented:
> Thanks for the PR!
> 
> Code coverage error is related - these lines aren't tested, but I think codecover is only picking them up now b/c they haven't been changed since we introduced code cover.
> 
> Would you be interested in writing tests to check that these error messages get triggered?

Yes, added. Thanks.
